ocaml/xapi/xapi_stunnel_server.ml

```diff
@@ -49,67 +49,71 @@ end = struct
-             ; "socket = r:SO_KEEPALIVE=1"
-             ; "socket = a:SO_KEEPALIVE=1"
-             ; "TIMEOUTclose = 1"
-             ; ( match Sys.getenv_opt "STUNNEL_IDLE_TIMEOUT" with
-               | None ->
-                   "; no idle timeout"
-               | Some x ->
-                   Printf.sprintf "TIMEOUTidle = %s" x
-               )
-             ; Stunnel.debug_conf_of_env ()
-             ; "protocol = proxy"
-               (* tells stunnel to include inet address info *)
-             ; ""
-             ; "[xapi]"
-             ; Printf.sprintf "accept = %s" accept
-             ; Printf.sprintf "cert = %s" cert
-             ; Printf.sprintf "ciphers = %s" Xcp_const.good_ciphersuites
-             ; "curve = secp384r1"
-             ; "options = CIPHER_SERVER_PREFERENCE"
-             ; "sslVersion = TLSv1.2"
-             ]
-           ; ( match client_auth_name with
-             | None ->
-                 (* No client certificate auth: forward to xapi on port 80
-                    for authentication by username+password or session/token. *)
-                 ["connect = 80"]
-             | Some name ->
-                 (* Verify the client certificate wrt the stored CA certs and name.
-                    Connect to xapi's dedicated unix socket if the cert is accepted,
-                    and no further auth is needed. Otherwise, redirect to port 80
-                    as above. *)
-                 [
-                   Printf.sprintf "connect = %s"
-                     Xapi_globs.unix_domain_socket_clientcert
-                 ; "redirect = 80"
-                 ; "verifyChain = yes"
-                 ; Printf.sprintf "CApath = %s" !Xapi_globs.trusted_certs_dir
-                 ; Printf.sprintf "checkHost = %s" name
-                 ]
-             )
-           ; [
-               ""
-             ; "# xapi connections use SNI 'pool' to request a cert"
-             ; "# the options from [xapi] are inherited unless overridden below"
-             ; "[pool]"
-             ; "connect = 80"
-             ; "sni = xapi:pool"
-             ; Printf.sprintf "cert = %s" pool_cert
-             ]
-           ]
+      let open Printf in
+      let cipher_options =
+        [
+          sprintf "ciphers = %s" Xcp_const.good_ciphersuites
+        ; "curve = secp384r1"
+        ; "options = CIPHER_SERVER_PREFERENCE"
+        ; "sslVersion = TLSv1.2"
+        ]
+      in
+      [
+        [
+          "; autogenerated by xapi"
+        ; sprintf "fips = %s" fips
+        ; "pid = /var/run/xapissl.pid"
+        ; "socket = r:TCP_NODELAY=1"
+        ; "socket = a:TCP_NODELAY=1"
+        ; "socket = l:TCP_NODELAY=1"
+        ; "socket = r:SO_KEEPALIVE=1"
+        ; "socket = a:SO_KEEPALIVE=1"
+        ; "TIMEOUTclose = 1"
+        ; ( match Sys.getenv_opt "STUNNEL_IDLE_TIMEOUT" with
+          | None ->
+              "; no idle timeout"
+          | Some x ->
+              sprintf "TIMEOUTidle = %s" x
```

Comment:
Should we check that this has an expected format? Do we expect this to be a number?

Reply:
It should be a number, yes. It's not new code though, and I don't really want to change its behaviour in this PR.